lib/cext/preprocess.rb

```diff
@@ -55,6 +55,8 @@ def preprocess(line)
 end
 
 if __FILE__ == $0
+  puts "# 1 \"#{ARGF.filename}\""
```

Comment:
Ah maybe with the `#line` directive? https://gcc.gnu.org/onlinedocs/cpp/Line-Control.html
Sounds like a great idea.